### PR TITLE
Better workaround for devices with 3G/LTE returning false when calling SystemFeature(PackageManager.FEATURE_TELEPHONY)

### PR DESCRIPTION
**Pull request by marco-scavuzzo**

### CODE REVIEW

**Comment by marco-scavuzzo** [comment]
Tested on two Nexus 7 with and without SIM card support. Now it behaves properly

File: app/src/main/java/eu/faircode/netguard/Util.java [hunk 1]
```diff
@@ -85,7 +85,8 @@ public static int getSelfVersionCode(Context context) {
     public static boolean hasTelephony(Context context) {
         PackageManager pm = context.getPackageManager();
         return pm.hasSystemFeature(PackageManager.FEATURE_TELEPHONY) ||
-                context.getSystemService(Context.TELEPHONY_SERVICE)!=null;
+                ((TelephonyManager) context.getSystemService(Context.TELEPHONY_SERVICE)).getNetworkType()!=
```
Comment (M66B) [inline app/src/main/java/eu/faircode/netguard/Util.java]: This can crash of get system service returns null.

Comment (marco-scavuzzo) [inline app/src/main/java/eu/faircode/netguard/Util.java]: True, I am going to fix it

**Comment by marco-scavuzzo** [comment]
Now it should be fine

**Comment by M66B** [comment]
@marco-scavuzzo can you please check my way of fixing and maybe review the commit?

**Comment by marco-scavuzzo** [comment]
Sure, I'll let you know asap

**Comment by marco-scavuzzo** [comment]
Seems good to me.
Thanks for fixing :)

**Comment by M66B** [comment]
Correction: thank you for fixing this :-)

**Comment by marco-scavuzzo** [comment]
Of course the workaround will not work if there's no mobile network (or the tablet is in air plane mode) and NetGuard was not previously started. So I was trying to see if there is another way around, that doesn't involve network connectivity at application start

**Comment by marco-scavuzzo** [comment]
Seems that getting the [SIM state](http://developer.android.com/reference/android/telephony/TelephonyManager.html#getSimState()) should be a more general solution. If it doesn't return ERROR or UNKNOWN we are sure that there is at least a sim card slot.
At least in Nexus 7 LTE  it always returns SIM_STATE_PUK_REQUIRED, whereas in the Nexus 7 without telephony is returns SIM_STATE_UNKNOWN.
If you agree, I can change this and make a pull request

**Comment by M66B** [comment]
IMHO the telephony manager workaround is better, because telephony might not always require a SIM card.

**Comment by marco-scavuzzo** [comment]
Yes, but lets consider the following scenario:  someone installs NetGuard on a Wi-Fi connection and, for some reason, there's no mobile network at all. In this case, no mobile network icons will be shown.  Then, he/she selects "Block all applications by default" and starts configuring NetGuard. When he/she will switch to a mobile network connection (when available) all applications will be blocked and he/she won't be able to fix this, unless he/she disables NetGuard, terminates the application and restarts it. Only at that point, if a mobile network is available, he/she will be able to change the settings for the mobile network.

**Comment by M66B** [comment]
This is only if the feature check returns a wrong result and the workaround kicks in, so I do not consider this as a big problem.

I am wondering why the feature check returns a wrong result.

**Comment by marco-scavuzzo** [comment]
Really I don't know. I would have understood it if I was using a custom
ROM, but I am on an original unrooted Android 6 version.
Il 19 dic 2015 9:45 PM, "Marcel Bokhorst" <notifications@github.com> ha
scritto:

**Pull request closed without merging**